examples/chat_history.py

```diff
@@ -1,49 +1,106 @@
 """A basic example showcasing how to manually manage chat history."""
 import os
 
-from mirascope import BasePrompt, OpenAICallParams, OpenAIChat
-from mirascope.base import Message
+from openai.types.chat import (
+    ChatCompletionAssistantMessageParam,
+    ChatCompletionMessageParam,
+    ChatCompletionSystemMessageParam,
+    ChatCompletionUserMessageParam,
+)
+
+from mirascope import OpenAICallParams
+from mirascope.openai import OpenAIPrompt
 
-os.environ["OPENAI_API_KEY"] = "YOUR_API_KEY"
+os.environ["OPENAI_API_KEY"] = "YOUR_OPENAI_API_KEY"
 
 
-class ChatPrompt(BasePrompt):
+class ChatPrompt(OpenAIPrompt):
     """A chat with history"""
 
     message: str
-    history: list[Message] = []
+    history: list[ChatCompletionMessageParam] = []
 
     call_params = OpenAICallParams(model="gpt-3.5-turbo")
 
     @property
-    def messages(self) -> list[Message]:
+    def messages(self) -> list[ChatCompletionMessageParam]:
         return [
-            {"role": "system", "content": "You are a helpful AI."},
```

Comment:
you should be able to leave these as dictionaries, which is what people using OpenAI messages will expect. No need to use their strict typing.

Reply:
Resolved